tracer/test/Datadog.Trace.Security.Unit.Tests/SpanCounterKeeperTests.cs

```diff
@@ -0,0 +1,62 @@
+// <copyright file="SpanCounterKeeperTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Security.Unit.Tests
+{
+    public class SpanCounterKeeperTests
+    {
+        [Fact]
+        public void FullRateLimitScenario()
+        {
+            var tracer = Mock.Of<IDatadogTracer>();
+
+            var limit = 100;
+            var rateLimiter = new SpanCounterKeeper(limit, true);
+
+            SingleRateLimitScenario(tracer, limit, rateLimiter);
+
+            rateLimiter.Reset();
+
+            SingleRateLimitScenario(tracer, limit, rateLimiter);
+        }
+
+        private static void SingleRateLimitScenario(IDatadogTracer tracer, int limit, SpanCounterKeeper rateLimiter)
+        {
+            for (var i = 0; i < limit; i++)
+            {
+                RunSpanTest(tracer, rateLimiter, SamplingPriorityValues.UserKeep);
+            }
+
+            RunSpanTest(tracer, rateLimiter, null, 1.0);
```

Comment:
LGTM, nit, maybe l.42 here could add a bit more testing like:
```

            for (var i = 1; i <= 50; i++)
            {
                RunSpanTest(tracer, rateLimiter, null, i);
            }
```

Reply:
These tests where removed when moving over to the existing tracer rate limiter.